src/java.base/share/classes/java/lang/Class.java

```diff
@@ -4366,49 +4405,70 @@ public String descriptorString() {
      *           may be removed in a future release, or upgraded to permanent
      *           features of the Java language.}
      *
-     * Returns an array containing {@code ClassDesc} objects representing all the
-     * direct subclasses or direct implementation classes permitted to extend or
+     * Returns an array containing {@code Class} objects representing the
+     * direct subinterfaces or subclasses permitted to extend or
      * implement this class or interface if it is sealed. The order of such elements
      * is unspecified. If this {@code Class} object represents a primitive type,
      * {@code void}, an array type, or a class or interface that is not sealed,
      * an empty array is returned.
      *
-     * @return an array of class descriptors of all the permitted subclasses of this class or interface
+     * The names of the permitted interfaces and classes are determined by
+     * reading the {@code PermittedSubclasses} class file attribute for this class.
+     * Then these names are converted to {@code Class} instances using
+     * {@linkplain #getClassLoader() the defining class loader} of the current
+     * {@code Class} object. If a name cannot be converted to the {@code Class}
+     * instance, it is silently excluded from the result.
```

Comment:
I think this paragraph will need a little bit of wordsmithing. The 3rd paragraph of getNestMembers might be useful to examine as it more clearly describes how the method attempts to "obtain" the Class object for each of the class names in the NestMembers attribute and maybe some of that wording could be used instead of using the term "convert".

Minor nit but the prevailing style for the @throws SecurityException is to align the description with the exception, probably best to keep it consistent if you can.

Reply:
Thanks, I've tried to improve the javadoc here:
https://github.com/openjdk/jdk/pull/1483/commits/4d484179e6e4d64ed460b997d25b4dca5d964016